cmd/root.go

```diff
@@ -211,3 +224,54 @@ func detectFormat(flags *pflag.FlagSet, origReader io.Reader) (utils.ContentType
 
 	return utils.ContentXml, reader
 }
+
+func processAsJSON(flags *pflag.FlagSet, reader io.Reader, w io.Writer, contentType utils.ContentType) error {
+	var (
+		jsonCompact bool
+		jsonDepth   int
+		result      interface{}
+	)
+	jsonCompact, _ = flags.GetBool("compact")
+	if flags.Changed("depth") {
+		jsonDepth, _ = flags.GetInt("depth")
+	} else {
+		jsonDepth = -1
+	}
+
+	switch contentType {
+	case utils.ContentXml, utils.ContentHtml:
+		doc, err := xmlquery.Parse(reader)
```

Comment:
Well, I took a little bit deeper look into the problem, and it seems like it's not as straightforward as for XML. Tags with the same names should be converted into arrays and it will lead to buffering that can ruin the whole idea. So probably, it's ok to go with your proposed solution, and maybe later, I'll try to experiment with a streaming parser.

Reply:
supportive of that